src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            T[] tempArray = (T[]) new Object[(int) (array.length * 1.5)];
```

Comment:
1.5 is a magic number
```suggestion
            T[] tempArray = (T[]) new Object[(int) (array.length * RESIZE_FACTOR)];
```